pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1434,13 +1436,22 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claim *v1.Persis
 		return
 	}
 
+	var selectedNode *v1.Node = nil
+	if nodeName, ok := claim.Annotations[annSelectedNode]; ok {
```

Comment:
Can you add a feature gate around this too?  Reason is to handle scenarios where you may want to turn the feature off, but the objects still have the annotation set.

Reply:
Ah, I see what you mean, the feature may be enabled at the beginning, and then disabled halfway.
I meant to make the code looks cleaner, but I'd say this indeed can happen. Will add it.

Reply:
added